pkg/kubelet/cm/deviceplugin/endpoint_store.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deviceplugin
+
+import (
+	"sync"
+)
+
+type store interface {
+	Get(resourceName string) (*endpoint, bool)
```

Comment:
If this is a intermediate to be replaced by sync.Map, how about to let this return an `interface{}` instead?

Reply:
The idea is to replace the underlying implementation storage by a sync.Map (i.e: `endpointStore` here)  :)

That allows the type casting (and the error checks) to be done at the `endpointStore` level rather than handling that at the manager level.

Reply:
Should this PR use sync.Map now straightway as @nqn shared on slack, https://groups.google.com/forum/#!topic/kubernetes-dev/INeuA9Flcb8